src/descriptor/tr.rs

```diff
@@ -236,9 +236,13 @@ impl<Pk: MiniscriptKey> Tr<Pk> {
             let mut builder = TaprootBuilder::new();
             for (depth, ms) in self.iter_scripts() {
                 let script = ms.encode();
-                builder = builder
-                    .add_leaf(depth, script)
-                    .expect("Computing spend data on a valid Tree should always succeed");
+                if let Ok(taproot_builder) = builder.add_leaf(depth, script) {
+                    builder = taproot_builder;
+                } else {
+                    return Err(Error::Unexpected(String::from(
+                        "Computing spend data on a valid Tree should always succeed",
+                    )));
```

Comment:
How do you actually hit this case? How do you obtain a `Tr` with an invalid tree? Can you add a unit test?

Relatedly, can you separate out the changes to `src/` into a separate commit from the changes to the integration test?

Thanks for your contribution!!

Reply:
1. This case is hit when the height of the tree is more than 128. In the Impl for `TaprootBuilder` in file taproot.rs of the rust-bitcoin project, there is a function `insert`, which returns a `TaprootBuilderError`.
2. For which function do you want me to add the unit tests?
3. Ok I will split it into two commits.

Reply:
Can you add a unit test for `Taproot::spend_info` which causes this new error path to be taken?

I think that would answer my "how do you hit this case" question, which I'm still confused about, sorry.

And thanks for splitting up the commits!

Reply:
Yep, I am also curious when this gets hit. This seems like a big bug if we need to change this.